src/ray/stats/metric_exporter_grpc_test.cc

```diff
@@ -149,6 +144,278 @@ TEST(MetricPointExporterTest, adds_global_tags_to_grpc) {
+    //   - Exporting 4 time-series
+    //   - 2 RPC payloads should be sent (1 payload will be taking ~180 bytes, it'll be
+    //   split in 2)
+    //
+    RAY_LOG(INFO) << "Test #1";
+
+    size_t kBatchSize = 4;
+    size_t maxPayloadSize = 250;
+    // Initialize the exporter
+    auto mockClient = std::make_shared<MockMetricsAgentClient>();
+    OpenCensusProtoExporter ocProtoExporter(
+        mockClient, WorkerID::Nil(), kBatchSize, maxPayloadSize);
+
+    rpc::ReportOCMetricsRequest proto;
+
+    ocProtoExporter.ExportViewData({
+        {view_descriptor, view_data},
+    });
+
+    ASSERT_THAT(mockClient->CollectedReportMetricsRequests().size(), 0);
+
+    auto requests = mockClient->CollectedReportOCMetricsRequests();
+
+    RAY_LOG(INFO) << "Request payload: " << requests[0].DebugString();
+
+    ASSERT_THAT(requests.size(), 2);
+    for (int i = 0; i < 2; ++i) {
+      // Both batches have to have 1 metric with 2 time-series each
+      auto metrics = requests[i].metrics();
+      ASSERT_THAT(metrics.size(), 1);
+      ASSERT_THAT(metrics[0].timeseries().size(), 2);
+    }
+  }
+
+  {
+    //
+    // Test #2: Splitting time-series across 6 batches (overflows payload size)
+    //   - Batch-size is 6, max-payload size is 250 (1 metric def + 2 time-series)
+    //   - Exporting 12 time-series
+    //   - 6 RPC payloads should be sent (since 1 payload will be taking ~250 bytes, it'll
+    //   be split in 6)
+    //
+    RAY_LOG(INFO) << "Test #2";
+
+    size_t kBatchSize = 6;
+    size_t maxPayloadSize = 250;  // 50% of the expected target payload size
+    // Initialize the exporter
+    auto mockClient = std::make_shared<MockMetricsAgentClient>();
+    OpenCensusProtoExporter ocProtoExporter(
+        mockClient, WorkerID::Nil(), kBatchSize, maxPayloadSize);
+
+    rpc::ReportOCMetricsRequest proto;
+
+    ocProtoExporter.ExportViewData({// NOTE: To avoid excessive boilerplate we just feed
+                                    // in same metrics to simulate larger batches
+                                    {view_descriptor, view_data},
+                                    {view_descriptor, view_data},
+                                    {view_descriptor, view_data}});
+
+    ASSERT_THAT(mockClient->CollectedReportMetricsRequests().size(), 0);
+
+    auto requests = mockClient->CollectedReportOCMetricsRequests();
+
+    RAY_LOG(INFO) << "Request payload: " << requests[0].DebugString();
+    RAY_LOG(INFO) << "Request payload: " << requests[1].DebugString();
+    RAY_LOG(INFO) << "Request payload: " << requests[2].DebugString();
+
+    ASSERT_THAT(requests.size(), 6);
+    for (int i = 0; i < 6; ++i) {
+      // Each of the batches have to have 1 metric with 2 time-series each
+      auto metrics = requests[i].metrics();
+      // ASSERT_THAT(metrics.size(), 1);
+      ASSERT_THAT(metrics[0].timeseries().size(), 2);
+    }
+  }
+
+  {
+    //
+    // Test #3: Splitting time-series across 1 batches (no overflowing)
```

Comment:
do we have a test case where there's overflow of size and also batches? If not, can you add a simple one?

Reply:
Yes, so both cases are covered in `export_view_data_split_by_batch_size` and `export_view_data_split_by_payload_size` respectively